src/ident/Ident.cc

```diff
@@ -243,9 +243,9 @@ Ident::Start(const Comm::ConnectionPointer &conn, IDCB * callback, void *data)
 
     conn->local.toUrl(key1, IDENT_KEY_SZ);
     conn->remote.toUrl(key2, IDENT_KEY_SZ);
-    const auto res = snprintf(key, sizeof(key), "%s,%s", key1, key2);
+    int res = snprintf(key, sizeof(key), "%s,%s", key1, key2);
     assert(res > 0);
-    assert(static_cast<std::make_unsigned<decltype(res)>::type>(res) < sizeof(key));
+    assert(static_cast<unsigned int>(res) < sizeof(key));
```

Comment:
Did not we add some code to strip `const` without losing `auto`? IIRC, you and I worked on that in this or another PR. Do you recall where we did that?

There are other PR places where `auto` got lost for similar reasons.

Reply:
Yes, that was on fixing make_unsigned (we had a make_const_somethingsomething).

IIRC think the type magic there is to take a const integer type and make it a const unsigned same integer type.

However in this case it feels like overengineering. The return value from snprintf is very well defined, it's int. Given that, going through type magic to make it const unsigned, only to have const-ness ignored (silently by default, verbosely with Wextra) makes the intent less clear and feels like needless complication

Reply:
Please note that you are not only losing `auto` here. You are also removing `const`. Yes, none of those things seem important when looking at two trivial lines of code, but there are good reasons for using core C++ features that prevent bugs and make code easier to maintain long-term, especially when they are used consistently. Moreover, once you start (carefully) removing const/auto, the next developer will do so with even fewer excuses.

However, I do not have the time to fight for this now, so I am resolving this change request.